R/as.data.table.R

```diff
@@ -241,17 +242,22 @@ as.data.table.data.frame = function(x, keep.rownames=FALSE, key=NULL, ...) {
   # fix for #1078 and #1128, see .resetclass() for explanation.
   setattr(ans, "class", .resetclass(x, "data.frame"))
   setalloccol(ans)
-  setkeyv(ans, key)
+  if (!is.null(key)) setkeyv(ans, key)
   ans
 }
 
-as.data.table.data.table = function(x, ...) {
+as.data.table.data.table = function(x, keep.rownames = FALSE, key = NULL, ...) {
```

Comment:
Given a default value to `key` might introduce breaking behavior.

Ideally `as.data.table.data.table` follows the following simple rules:
`as.data.table.data.table(keyed-dt)` returns a keyed data.table
`as.data.table.data.table(unkeyed-dt` returns a data.table without a key